flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/datastream/IterativeDataStream.java

```diff
@@ -201,14 +204,16 @@ public ConnectedIterativeDataStream(IterativeDataStream<I> input, TypeInformatio
 		 * @return The feedback stream.
 		 * 
 		 */
+		@SuppressWarnings({ "rawtypes", "unchecked" })
 		public DataStream<F> closeWith(DataStream<F> feedbackStream) {
-			DataStream<F> iterationSink = new DataStreamSink<F>(input.environment, "Iteration Sink",
-					null, null);
+			if (input.closed) {
+				throw new IllegalStateException(
```

Comment:
Since now we build the loops at the job graph generation, wouldn't it be reasonable to do an implicit union there?